clang/lib/Serialization/ASTReader.cpp

```diff
@@ -3349,10 +3351,29 @@ ASTReader::ReadControlBlock(ModuleFile &F,
                                       .getModuleCache()
                                       .getInMemoryModuleCache()
                                       .isPCMFinal(F.FileName);
-      if (isDiagnosedResult(Result, Capabilities) || recompilingFinalized)
+      if (isDiagnosedResult(Result, Capabilities) || recompilingFinalized) {
```

Comment:
Can you clarify why you're conditionalizing this diagnostic on these?

Reply:
It's because this is where fatal `OutOfDate` errors are diagnosed.

Reply:
Is it? `isDiagnosedResult(Result, Capabilities)` returns `true` for other cases, not only `OutOfDate` errors.